engine/explain_test.go

```diff
@@ -186,39 +187,83 @@ func TestAnalyzeOutputNode_Samples(t *testing.T) {
 	ctx := context.Background()
 
 	load := `load 30s
-				http_requests_total{pod="nginx-1"} 1+1x10
-				http_requests_total{pod="nginx-2"} 1+2x14`
+				http_requests_total{pod="nginx-1"} 1+1x100
+				http_requests_total{pod="nginx-2"} 1+1x100`
 
 	tstorage := promql.LoadedStorage(t, load)
 	defer tstorage.Close()
+	minT := tstorage.Head().Meta().MinTime
+	maxT := tstorage.Head().Meta().MaxTime
+
+	query, err := ng.NewInstantQuery(ctx, tstorage, nil, "http_requests_total", time.Unix(0, 0))
+	testutil.Ok(t, err)
+	queryResults := query.Exec(context.Background())
+	testutil.Ok(t, queryResults.Err)
 
 	rangeQry, err := ng.NewRangeQuery(
 		ctx,
 		tstorage,
 		promql.NewPrometheusQueryOpts(false, 0),
-		"sum(rate(http_requests_total[1m])) by (pod)",
-		time.Unix(0, 0),
-		time.Unix(2*60*60, 0),
+		"sum(rate(http_requests_total[10m])) by (pod)", // Increase range to 60 minutes
+		time.Unix(minT, 0),
+		time.Unix(maxT, 0),
 		60*time.Second,
 	)
 	testutil.Ok(t, err)
-
-	queryResults := rangeQry.Exec(context.Background())
+	queryResults = rangeQry.Exec(context.Background())
 	testutil.Ok(t, queryResults.Err)
 
 	explainableQuery := rangeQry.(engine.ExplainableQuery)
 	analyzeOutput := explainableQuery.Analyze()
 	require.Greater(t, analyzeOutput.PeakSamples(), int64(0))
 	require.Greater(t, analyzeOutput.TotalSamples(), int64(0))
 
-	query, err := ng.NewInstantQuery(ctx, tstorage, nil, "http_requests_total", time.Unix(0, 0))
-	testutil.Ok(t, err)
-
-	queryResults = query.Exec(context.Background())
-	testutil.Ok(t, queryResults.Err)
-
-	explainableQuery = query.(engine.ExplainableQuery)
+	explainableQuery = rangeQry.(engine.ExplainableQuery)
 	analyzeOutput = explainableQuery.Analyze()
 	require.Greater(t, analyzeOutput.PeakSamples(), int64(0))
 	require.Greater(t, analyzeOutput.TotalSamples(), int64(0))
+	result := renderAnalysisTree(analyzeOutput, 0)
+	expected := `[duplicateLabelCheck]: 0 peak: 0
+|---[concurrent(buff=2)]: 0 peak: 0
+|   |---[aggregate] sum by ([pod]): 0 peak: 0
+|   |   |---[duplicateLabelCheck]: 0 peak: 0
+|   |   |   |---[coalesce]: 0 peak: 0
```

Comment:
Why do we get 5 shards here? Would this test depend on how many cores a machine has?

Reply:
I think amount shards should come from query options with default taken from the current "number of cores". That way we could just set it for the test to something lower.

Reply:
As long as we do it in a backwards compatible way, and we don't blow up the scope of the PR, I would be fine with that change.